### DISCUSSION TITLE
Teams: Must 'team_id' not be null on pivot table?

### DISCUSSION THREAD

**Discussion by dsturm** [spatie/laravel-permission#1840] [Q&A] [answered]
Hej.

I set up a new `laravel/jestream` app, enabled teams and now installed `spatie/laravel-permission` with active team support. After this, I've created a global role (with `team_id` null) and wanted to assign this role to a user. But since the migration for the table `model_has_roles` sets the `team_id` as not nullable I run into a QueryException.

```sql
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'team_id' cannot be null (SQL: insert into `model_has_roles` (`model_id`, `model_type`, `role_id`, `team_id`) values (1, App\Models\User, 1, ?))
```

Since I did not find any open or closed issues on this: Do I miss something? Shouldn't we are able to omit `team_id` for global roles? 🤔 

Best

**Comment by erikn69** [comment, marked as answer]
Automatic Global Role SuperAdmin
-------
Global roles can be assigned to different teams, and as the primary key of the relationship this is required, If you have a global user, when you register a new team you must assign it the global default role(SuperAdmin) by created event , example on your team model
```php
public static function boot(){
    parent::boot();
    // here assign this team to a global user with global default role
    self::created(function($teamModel){
        // get session team_id to restore it later (maybe optional)
       $session_team_id = getPermissionsTeamId();
       setPermissionsTeamId($teamModel->your_team_id);
       User::find('user_super_admin_id')->assignRole('SuperAdminGlobalExample')
       setPermissionsTeamId($session_team_id);
    }
});
```

Working on session
------
First, you must register the id of the team that you are going to use on the session, look at the documentation [here](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#working-with-teams-permissions),

so, on you middleware you needs 
```php
//NOTE: this can be on the session middleware or other, 
//      but always before starting to use this package like `$user->hasRoles('role');`
setPermissionsTeamId(session('your_team_id'));
```
Or, if you are going to assign roles to a different team than the one in the session, you must first put the desired team with the same function(as in the first example)
```php
setPermissionsTeamId($teamModel->your_team_id);
```

**Reply by PaolaRuby** [reply]
Thanks, that is what i was looking for

**Reply by dsturm** [reply]
Hej @erikn69, thanks for your answer!

**Reply by erikn69** [reply]
@dsturm if you have questions or suggestions about teams just tag me, it is really new feature, still needs work, but at least it is an option to use teams
